pkg/volumeclaimbinder/persistent_volume_claim_binder.go

```diff
@@ -237,32 +230,52 @@ func syncClaim(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCli
 			return fmt.Errorf("A volume match does not exist for persistent claim: %s", claim.Name)
 		}
 
-		claimRef, err := api.GetReference(claim)
-		if err != nil {
-			return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
-		}
-
 		// make a binding reference to the claim.
-		// triggers update of the volume in this controller, which builds claim status
-		volume.Spec.ClaimRef = claimRef
-		volume, err = binderClient.UpdatePersistentVolume(volume)
+		// triggers update of the claim in this controller, which builds claim status
+		claim.Spec.VolumeName = volume.Name
+		// TODO: make this similar to Pod's binding both with BindingREST subresource and GuaranteedUpdate helper in etcd.go
+		claim, err = binderClient.UpdatePersistentVolumeClaim(claim)
 		if err == nil {
 			nextPhase = api.ClaimBound
-		}
-		if err != nil {
+			glog.V(5).Infof("PersistentVolumeClaime[%s] is bound\n", claim.Name)
+		} else {
 			// Rollback by unsetting the ClaimRef on the volume pointer.
 			// the volume in the index will be unbound again and ready to be matched.
-			volume.Spec.ClaimRef = nil
+			claim.Spec.VolumeName = ""
 			// Rollback by restoring original phase to claim pointer
 			nextPhase = api.ClaimPending
 			return fmt.Errorf("Error updating volume: %+v\n", err)
 		}
 
-	// bound claims requires no maintenance.  Deletion by the user is the last lifecycle phase.
 	case api.ClaimBound:
-		// This is the end of a claim's lifecycle.
-		// After claim deletion, a volume is recycled when it verifies its claim is unbound
-		glog.V(5).Infof("PersistentVolumeClaime[%s] is bound\n", claim.Name)
+		volume, err := binderClient.GetPersistentVolume(claim.Spec.VolumeName)
+		if err != nil {
+			return fmt.Errorf("Unexpected error getting persistent volume: %v\n", err)
+		}
+
+		if volume.Spec.ClaimRef == nil {
+			claimRef, err := api.GetReference(claim)
+			if err != nil {
+				return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
+			}
+			volume.Spec.ClaimRef = claimRef
+			_, err = binderClient.UpdatePersistentVolume(volume)
+			if err != nil {
+				return fmt.Errorf("Unexpected error saving PersistentVolume.Status: %+v", err)
+			}
+		}
+
+		if len(claim.Status.AccessModes) == 0 {
```

Comment:
I moved the comment for transferring "actuals" to this line and added a bit more that states checking for these accessModes is sufficient to know if we've already set Status on PVC or not.